fmlloader/src/main/java/net/minecraftforge/fml/loading/targets/ForgeGametestDevLaunchHandler.java

```diff
@@ -0,0 +1,20 @@
+package net.minecraftforge.fml.loading.targets;
+
+import net.minecraftforge.api.distmarker.Dist;
+
+import java.util.concurrent.Callable;
+
+public class ForgeGametestDevLaunchHandler extends CommonDevLaunchHandler {
+    @Override public String name() { return "forgegametestdev"; }
+    @Override public Dist getDist() { return Dist.DEDICATED_SERVER; }
```

Comment:
To answer number 1, the inheritance of both classes is different. This is how all the launch handler works. It could be potentially refactored in another PR, but isn't something urgent as it's just a few lines of code duplication.
For number 2, I have no idea if the gametest **server** can test client stuff. `GameTestServer` is just a subclass of `MinecraftServer`, but I don't think it has access to any client stuff. However, one could theoretically write a game test and register it only on Dist.CLIENT, assuming they have `forge.gametest` set to `true`, then they can run that test manually with `/test runall` in their client instance.

Reply:
After having had a chance to actually do some testing and take a better/more in depth look at how game tests work I better understand that the reason it is dedicated server is that is just what the environment that the runs that are being tested are for in that it launches a server and runs the test, and that for testing the client launching the client and using the command works fine.

Part of me wishes there was a better solution for how to register tests that are "client side only" rather than having the modders have to do the dist checks themselves, but that is not that big of a deal and if someone comes up with a better way of doing that then they can always make a follow up PR. Thinking about it maybe the "best" solution from an API standpoint might be to allow for a dist value to be specified in the `GameTest` and `GameTestGenerator` annotations (and maybe even BeforeBatch and AfterBatch) and then patch the if statements that do things if the annotation was found to also require the dist to be correct... But given how forge has to patch things I feel like that would probably be overkill from extra maintenance standpoint, compared to just telling devs to put the client only tests in a different class and then manually check the dist and only add the client only test classes if it is the client.